flake.nix

```diff
@@ -0,0 +1,123 @@
+{
+  description =
+    "tinygrad: For something between PyTorch and karpathy/micrograd. Maintained by tiny corp.";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils }@I:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        P = import nixpkgs {
+          inherit system;
+          config.allowUnfree = true;
+        };
+        B = builtins;
+        L = P.lib;
+
+        pythonVersion = "310";
+        pythonPackageStr = "python${pythonVersion}";
+        pythonNixpkgs = P."${pythonPackageStr}Packages";
+
+        tinygrad = pythonNixpkgs.buildPythonPackage {
+          pname = "tinygrad";
+          version = "v.0.6.0";
+          src = ./.;
+
+          propagatedBuildInputs = with pythonNixpkgs; [
+            networkx
+            numpy
+            pillow
+            pyopencl
+            pyyaml
+            requests
+            tqdm
+          ];
+
+          # TODO(b7r6): enable when all tests pass...
+          doCheck = false;
+          pythonImportsCheck = [ "tinygrad" ];
+        };
+
+        commonBuildInputs = with P; [ git nixfmt ];
+        commonPythonPackages = pypkgs:
+          with pypkgs; [
+            flake8
+            mypy
+            onnx
+            onnxruntime
+            opencv4
+            pylint
+            pytest
+            pytest-xdist
+            python-lsp-server
```

Comment:
We should avoid pulling in lsp stuff in the devshell cuz peoples dev environments/editors may be different and may use different lsp implementations.

Reply:
Also a little subtle. I have `python-lsp-server` in my home manager, but my home manager python is `311` which can't work for `tinygrad` at the moment because `torch` deps on `tensorboard` which breaks on `311`.

I'll defer to your judgement here, but `python-lsp-server` seems in 2023 to be a comparably core dependency to e.g. `pytest` for people who still use like emacs and vim.

LMK how you want this resolved.

Reply:
`tensorboard` should be fixed on 3.11 with https://github.com/NixOS/nixpkgs/pull/227957. I myself do use neovim as my editor but I use pyright as a language server, so I'm not sure how `python-lsp-server` is a dependency for `pytest`. Is there a reason I'm not getting?